### PR TITLE
Opt for serving new spec JS/CSS dependencies locally instead of downloading from CDNs

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR does two things:

* Updates the git submodule for the new spec hugo theme ([google/docsy](https://github.com/google/docsy)) to our fork ([matrix-org/docsy](https://github.com/matrix-org/docsy)) which has a couple changes to load JS from local sources instead of remote, as well as allowing sites to override the URL that font CSS files are loaded from. Note that my definition of "font CSS" files here as CSS files that contain settings and point to locations of where font files (`.woff2`) are located.
* Adds a script (and the files generated as a result of running that script) that can take a google fonts URL, download the fonts it points to and spit out those fonts as well as a font CSS file pointing to them for local distribution. We then use the resulting font CSS file in our project variables.

This brings the benefit of not serving files from a CDN which can track users across the web, as well as inadvertently pinning docsy to a specific commit. The downside is that we need to remember to update [matrix-org/docsy](https://github.com/matrix-org/docsy) when needed (and apply the patches on top, though they're quite small).

### CODE REVIEW

File: static/css/fonts/Inter.css [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  font-weight: 700;
+  src: url(../../fonts/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuFuYAZBhiI2B.woff2) format('woff2');
+  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+1EA0-1EF9, U+20AB;
+}
+/* latin-ext */
+@font-face {
+  font-family: 'Inter';
+  font-style: normal;
+  font-weight: 700;
+  src: url(../../fonts/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuFuYAZFhiI2B.woff2) format('woff2');
```
Comment (turt2live) [inline static/css/fonts/Inter.css:158]: if it's not too awful, renaming these files to be `inter.woff2` and variants then that would be awesome.

**Review by turt2live — APPROVED** [review]
lgtm generally

File: static/css/fonts/Inter.css (hunk 1 above)
Comment (anoadragon453) [reply static/css/fonts/Inter.css:158]: Might be a little tricky as some only differ by unicode range, but perhaps I can use the text between the `/* ... */` to help make the filenames unique.

Comment (anoadragon453) [reply static/css/fonts/Inter.css]: Font filenames now look like `Inter-latin-ext-normal-400.woff2`, which should be a lot nicer :slightly_smiling_face:

**Pull request merged**